dashboard/src/components/Config/AppRepoList/AppRepoControl.tsx

```diff
@@ -12,28 +12,27 @@ import "./AppRepoControl.css";
 
 interface IAppRepoListItemProps {
   repo: IAppRepository;
-  namespace: string;
   kubeappsNamespace: string;
   secret?: ISecret;
+  refetchRepos: () => void;
 }
 
 export function AppRepoControl({
-  namespace,
   repo,
   secret,
   kubeappsNamespace,
+  refetchRepos,
 }: IAppRepoListItemProps) {
   const [modalIsOpen, setModalOpen] = useState(false);
   const [refreshing, setRefreshing] = useState(false);
   const openModal = () => setModalOpen(true);
   const closeModal = () => setModalOpen(false);
   const dispatch: ThunkDispatch<IStoreState, null, Action> = useDispatch();
 
-  const handleDeleteClick = (repoName: string, repoNamespace: string, currentNamespace: string) => {
+  const handleDeleteClick = (repoName: string, repoNamespace: string) => {
     return async () => {
       await dispatch(actions.repos.deleteRepo(repoName, repoNamespace));
-      // repoNamespace?
-      dispatch(actions.repos.fetchRepos(currentNamespace));
+      refetchRepos();
```

Comment:
the logic to re-fetch repos is in the AppList component so this component don't need to be aware of that complexity.

Reply:
You meant AppRepoList right? Anyway great, yes, this component only needs to know to call `refetchRepos()`.